src/plugins/discover/public/application/angular/context/api/utils/get_es_query_sort.ts

```diff
@@ -6,19 +6,30 @@
  * Side Public License, v 1.
  */
 
-import { EsQuerySortValue, SortDirection } from '../../../../../kibana_services';
+import {
+  EsQuerySortValue,
+  SortDirection,
+  SortDirectionFormat,
+} from '../../../../../kibana_services';
 
 /**
  * Returns `EsQuerySort` which is used to sort records in the ES query
  * https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-sort.html
  * @param timeField
  * @param tieBreakerField
  * @param sortDir
+ * @param nanos
  */
 export function getEsQuerySort(
   timeField: string,
   tieBreakerField: string,
-  sortDir: SortDirection
+  sortDir: SortDirection,
+  nanos?: string
 ): [EsQuerySortValue, EsQuerySortValue] {
-  return [{ [timeField]: sortDir }, { [tieBreakerField]: sortDir }];
+  const time = {} as Record<string, SortDirectionFormat>;
+  time[timeField] = {
+    order: sortDir,
+    format: nanos ? 'strict_date_optional_time_nanos' : 'strict_date_optional_time',
+  };
+  return [{ ...time }, { [tieBreakerField]: sortDir }];
```

Comment:
Now this can be simplified, and the import `SortDirectionFormat` can be removed:

```
return [
    {
      [timeField]: {
        order: sortDir,
        format: nanos ? 'strict_date_optional_time_nanos' : 'strict_date_optional_time',
      },
    },
    { [tieBreakerField]: sortDir },
  ];
```